packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -0,0 +1,143 @@
+import { TealError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { parseNum, toBytes } from "../lib/parse-data";
+import type { AppArgs, TEALStack } from "../types";
+import { Op } from "./opcode";
+
+const BIGINT0 = BigInt("0");
+// pops string([]byte) from stack and pushes it's length to stack
+export class Len extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 1);
+    const top = stack.pop();
+    if (typeof top === 'undefined' || !(top instanceof Uint8Array)) {
+      throw new TealError(ERRORS.TEAL.INVALID_OP_ARG, {
+        opcode: "len"
+      });
+    }
+    stack.push(BigInt(top.length));
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their sum(a + b) to stack
+// panics on overflow (result > max_unit64)
+export class Add extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    this.checkOverFlow(a + b);
+    stack.push(a + b);
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their diff(a - b) to stack
+// panics on underflow (result < 0)
+export class Sub extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    this.checkUnderFlow(a - b);
+    stack.push(a - b);
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their division(a / b) to stack
+// panics if b == 0
+export class Div extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (b === BIGINT0) {
+      throw new TealError(ERRORS.TEAL.INVALID_OP_ARG, {
+        opcode: "/"
+      });
+    }
+    stack.push(a / b);
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their mult(a * b) to stack
+// panics on overflow (result > max_unit64)
+export class Mul extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    this.checkOverFlow(a * b);
+    stack.push(a * b);
+  }
+}
+
+// pushes 1st argument from argument array to stack
+export class Arg_0 extends Op {
+  readonly _args;
+
+  constructor (args: AppArgs) {
+    super();
+    this._args = args;
+  };
+
+  execute (stack: TEALStack): void {
+    const arg = this._args[0];
+    if (arg === undefined || (typeof arg !== 'string' && typeof arg !== 'number')) {
+      throw new TealError(ERRORS.TEAL.INVALID_OP_ARG, {
+        opcode: "arg_0"
+      });
+    }
+    if (typeof arg === 'number') {
+      this.checkOverFlow(parseNum(arg));
+      stack.push(parseNum(arg));
+    } else { stack.push(toBytes(arg)); }
```

Comment:
I think we should disallow `number`. `arg` should have same type as `StackElem`: only `BigInt` or `Uint8Array`

Reply:
Ok. But, since the user is passing the arguments, is it good to ask the user to pass `bigint` instead of regular `int`?

Reply:
The input is a TEAL code - so everything is a text there. We will read that text an parse it accordingly.

Reply:
ah, important thing, I've just realized that args are ALWAYS `bytes`!

Reply:
https://developer.algorand.org/docs/reference/teal/opcodes/#arg

Reply:
Yes. :+1: Updating to `bytes`